### PR TITLE
Story/io 7/color code project cells

### PR DESCRIPTION
**Pull request by EngbergJonas**
- Add an active color for project cells that have either planning, construction or both during that year
- Create project cells from the correct values
- Only allow editing of cell values that are active

### CODE REVIEW

File: src/components/Planning/PlanningGroupsTable/PlanningGroupsTableRow.tsx [hunk 1]
```diff
@@ -3,24 +3,49 @@ import { planProjectValues } from '@/mocks/common';
...
+      type: getCellType(year),
+      objectKey: key,
+      projectId: project.id,
+    };
+  };
+
+  const cells = [];
 
-interface IPlanningProjectsTableProps {
+  for (const [key, value] of Object.entries(project)) {
```
Comment (kgunga) [inline src/components/Planning/PlanningGroupsTable/PlanningGroupsTableRow.tsx:35]: return Object.entries(project).map or flatMap

File: src/components/Planning/PlanningGroupsTable/PlanningGroupsTableRow.tsx [hunk 2]
```diff
@@ -29,13 +54,15 @@ interface IPlanningProjectsTableProps {
 /**
  * We're only mapping the project name here for now since the values aren't yet implemented
  */
-const PlanningGroupsTableRow: FC<IPlanningProjectsTableProps> = ({
+const PlanningGroupsTableRow: FC<IPlanningGroupsTableRowProps> = ({
```
Comment (kgunga) [inline src/components/Planning/PlanningGroupsTable/PlanningGroupsTableRow.tsx:57]: onClick={(e) => onProjectMenuClick(project.id, e as unknown as MouseEvent)} move this anonymous function into const onClick = useCallback((e) => ...., [project])

File: src/components/Planning/PlanningGroupsTable/ProjectCell.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        onBlur={handleBlur}
+        onFocus={handleFocus}
+        onChange={handleChange}
+        disabled={type === 'none'}
+      />
+    </td>
+  );
+};
+
+export default ProjectCell;
```
Comment (kgunga) [inline src/components/Planning/PlanningGroupsTable/ProjectCell.tsx:49]: Component receives properties so wrap with memo

File: src/utils/common.ts [hunk 4]
```diff
@@ -126,3 +126,21 @@ export const classesToOptions = (classes: Array<IClass>): Array<IOption> =>
...
+ * @returns boolean
+ */
+export const isInYearRange = (
+  year: number,
+  startDate: string | undefined,
+  endDate: string | undefined,
+): boolean => {
+  const startYear = parseInt(startDate?.split('.').pop() || '0');
+  const endYear = parseInt(endDate?.split('.').pop() || '0');
+  return startYear && endYear ? year >= startYear && year <= endYear : false;
```
Comment (kgunga) [inline src/utils/common.ts:145]: can we use here moment to check is year between start and end years?

**Review by kgunga — CHANGES_REQUESTED** [review]

File: src/utils/common.ts (hunk 4 above)
Comment (EngbergJonas) [reply src/utils/common.ts:145]: moment is added is the next feature https://github.com/City-of-Helsinki/infraohjelmointi-ui/pull/38.

File: src/components/Planning/PlanningGroupsTable/PlanningGroupsTableRow.tsx (hunk 1 above)
Comment (EngbergJonas) [reply src/components/Planning/PlanningGroupsTable/PlanningGroupsTableRow.tsx:35]: This behaviour changes in https://github.com/City-of-Helsinki/infraohjelmointi-ui/pull/38, could it be fixed there?

**Pull request merged**